core/templates/pages/exploration-editor-page/modal-templates/state-diff-modal.controller.spec.ts

```diff
@@ -0,0 +1,137 @@
+      StateObjectFactory = $injector.get('StateObjectFactory');
+
+      spyOn(CsrfTokenService, 'getTokenAsync')
+        .and.returnValue($q.resolve('sample-csrf-token'));
+
+      newState = StateObjectFactory.createDefaultState(newState);
+      oldState = StateObjectFactory.createDefaultState(oldState);
+
+      $uibModalInstance = jasmine.createSpyObj(
+        '$uibModalInstance', ['close', 'dismiss']);
+
+      // There are two post requests with the same endpoint.
+      $httpBackend.expectPOST('/createhandler/state_yaml/exp1').respond({
+        yaml: 'Yaml data'
+      });
+      $httpBackend.expectPOST('/createhandler/state_yaml/exp1').respond({
+        yaml: 'Yaml data'
+      });
+
+      $scope = $rootScope.$new();
+      $controller('StateDiffModalController', {
+        $scope: $scope,
+        $uibModalInstance: $uibModalInstance,
+        headers: headers,
+        newState: newState,
+        newStateName: newStateName,
+        oldState: oldState,
+        oldStateName: oldStateName
+      });
+    }));
+
+    it('should evaluate scope variables values correctly', function() {
+      expect($scope.headers).toBe(headers);
+      expect($scope.newStateName).toBe(newStateName);
+      expect($scope.oldStateName).toBe(oldStateName);
+      expect($scope.yamlStrs).toEqual({});
+    });
+
+    it('should evaluate yaml strings object', function() {
+      $httpBackend.flush();
+      expect($scope.yamlStrs.leftPane).toBe('Yaml data');
+      expect($scope.yamlStrs.rightPane).toBe('Yaml data');
+    });
+  });
+
+  describe('when new state and old state are falsy', function() {
+    newState = null;
+    oldState = null;
+
+    beforeEach(angular.mock.inject(function($injector, $controller) {
+      $q = $injector.get('$q');
+      var $rootScope = $injector.get('$rootScope');
+      $timeout = $injector.get('$timeout');
+      ContextService = $injector.get('ContextService');
+      spyOn(ContextService, 'getExplorationId').and.returnValue(explorationId);
+      CsrfTokenService = $injector.get('CsrfTokenService');
+
+      spyOn(CsrfTokenService, 'getTokenAsync')
+        .and.returnValue($q.resolve('sample-csrf-token'));
+
+      $uibModalInstance = jasmine.createSpyObj(
+        '$uibModalInstance', ['close', 'dismiss']);
+
+      $scope = $rootScope.$new();
+      $controller('StateDiffModalController', {
+        $scope: $scope,
+        $uibModalInstance: $uibModalInstance,
+        headers: headers,
+        newState: newState,
+        newStateName: newStateName,
+        oldState: oldState,
+        oldStateName: oldStateName
+      });
+    }));
+
+    it('should evaluate scope variables values correctly', function() {
+      expect($scope.headers).toBe(headers);
+      expect($scope.newStateName).toBe(newStateName);
+      expect($scope.oldStateName).toBe(oldStateName);
+      expect($scope.yamlStrs).toEqual({});
+    });
+
+    it('should evaluate yaml strings object', function() {
+      $timeout.flush(400);
```

Comment:
Add a comment explaining why this is needed.

Reply:
Done.